### ISSUE TITLE
Problems using with Android app

### ISSUE THREAD

**Issue by tmm1** [jwtk/jjwt#382]
The readme says to add this to gradle:

```
dependencies {
    compile 'io.jsonwebtoken:jjwt-api:0.10.2'
    runtime 'io.jsonwebtoken:jjwt-impl:0.10.2' 
    runtime('io.jsonwebtoken:jjwt-orgjson:0.10.2') {
        exclude group: 'org.json', module: 'json' //provided by Android natively
    }
    // Uncomment the next line if you want to use RSASSA-PSS (PS256, PS384, PS512) algorithms:
    //runtime 'org.bouncycastle:bcprov-jdk15on:1.60'
}
```

However this produces errors as `runtime` is not a valid function in the latest version of gradle.

If I change to `api` or `implementation`, I get proguard failures:

```
Warning: io.jsonwebtoken.io.OrgJsonSerializer: can't find referenced class org.json.JSONString
Warning: io.jsonwebtoken.io.OrgJsonSerializer: can't find referenced class org.json.JSONWriter
Warning: io.jsonwebtoken.io.OrgJsonSerializer: can't find referenced class org.json.JSONWriter
```

**Comment by lhazlewood** [comment]
Thanks for reporting this.  What version of Android Studio and/or Gradle are you using?

Those two org.json classes aren't needed on Android - what would be the ProGuard exclusions for them? I'd like to add this to documentation.

I don't know a whole lot about Android dependency scopes, but does `implementation` work for you?

**Comment by tmm1** [comment]
Using Android Studio 3.1.4

If I remove the `exclude` then I get this other error:

```
Error: json defines classes that conflict with classes now provided by Android. Solutions include finding newer versions or alternative libraries that don't have the same problem (for example, for httpclient use HttpUrlConnection or okhttp instead), or repackaging the library using something like jarjar. [DuplicatePlatformClasses]
    implementation 'io.jsonwebtoken:jjwt-api:0.10.2'
                       ~~~~
```

The `compile` in the gradle example should definitely be changed to `implementation`. But I'm not sure what `runtime` should be changed to, and so far I'm unable to make it work with my app..

**Comment by tmm1** [comment]
Okay I changed `runtime` to `api` and then added this to my .pro file:

```
-dontwarn org.json.*
```

Everything appears to work now.

**Comment by lhazlewood** [comment]
I just read this:

https://medium.com/mindorks/implementation-vs-api-in-gradle-3-0-494c817a6fa

You definitely don't want to use  `api` for runtime-only dependencies.  

Will this work?

```gradle
dependencies {
    api 'io.jsonwebtoken:jjwt-api:0.10.2'
    implementation 'io.jsonwebtoken:jjwt-impl:0.10.2' 
    implementation('io.jsonwebtoken:jjwt-orgjson:0.10.2') {
        exclude group: 'org.json', module: 'json' //provided by Android natively
    }
    // Uncomment the next line if you want to use RSASSA-PSS (PS256, PS384, PS512) algorithms:
    //implementation 'org.bouncycastle:bcprov-jdk15on:1.60'
}
```

with

`-dontwarn org.json.*` 

?

**Comment by lhazlewood** [comment]
Ah, I just found `runtimeOnly` in the Android documentation.  So it should probably be:

```gradle
dependencies {
    api 'io.jsonwebtoken:jjwt-api:0.10.2'
    runtimeOnly 'io.jsonwebtoken:jjwt-impl:0.10.2' 
    runtimeOnly('io.jsonwebtoken:jjwt-orgjson:0.10.2') {
        exclude group: 'org.json', module: 'json' //provided by Android natively
    }
    // Uncomment the next line if you want to use RSASSA-PSS (PS256, PS384, PS512) algorithms:
    //runtimeOnly 'org.bouncycastle:bcprov-jdk15on:1.60'
}
```

**Comment by lhazlewood** [comment]
Thanks for your help with this today!  I'll get the release out asap.

**Comment by tmm1** [comment]
Thanks. I don't think there's anything different between 0.10.2 and 0.10.3 as far as this issue is concerned. But it would be helpful if the README.md on master was updated as well, since that's where I copy/pasted the original incorrect configuration from.

**Comment by lhazlewood** [comment]
Yep, I'll merge to master right after the 0.10.3 release.  Branch patches/fixes always have to go into master too.  Thanks again!

**Comment by lhazlewood** [comment]
Released with 0.10.3 - please allow 30 min or so to propagate to Maven Central.
